mods/mc_teacher/callbacks.lua

```diff
@@ -406,11 +527,140 @@ minetest.register_on_player_receive_fields(function(player, formname, fields)
                     minetest.chat_send_player(pname, minetest.colorize(mc_core.col.log, "[Minetest Classroom] You can not ban yourself from the server."))
                 end
             end
+        elseif fields.p_teleport then
+            local pname = player:get_player_name()
+            local sel_pname = context.p_list[context.selected_p_player]
+            local sel_pobj = minetest.get_player_by_name(sel_pname or "")
+            if sel_pname and sel_pobj then
+                local destination = sel_pobj:get_pos()
+                local realm = Realm.GetRealmFromPlayer(sel_pobj)
+                if realm and realm:getCategory().joinable(realm, player) then
+                    realm:TeleportPlayer(player)
+                    player:set_pos(destination)
+                    minetest.chat_send_player(pname, minetest.colorize(mc_core.col.log, "[Minetest Classroom] Teleported to "..sel_pname.."!"))
+                else
+                    minetest.chat_send_player(pname, minetest.colorize(mc_core.col.log, "[Minetest Classroom] Could not teleport to "..sel_pname.."."))
+                end
+            else
+                minetest.chat_send_player(pname, minetest.colorize(mc_core.col.log, "[Minetest Classroom] Player "..tostring(sel_pname).." could not be found!"))
```

Comment:
I forced it to be a string to avoid the possibility of concatenating a `nil` element - not having a `tostring` there caused some crashes while testing